react/ActionMenu/index.jsx

```diff
@@ -1,54 +1,62 @@
-import React, { useCallback, useContext } from 'react'
+import React from 'react'
 import PropTypes from 'prop-types'
 import cx from 'classnames'
 import ClickAwayListener from '@material-ui/core/ClickAwayListener'
 import styles from './styles.styl'
 import { Media, Bd, Img } from '../Media'
 import BottomDrawer from '../BottomDrawer'
 import withBreakpoints from '../helpers/withBreakpoints'
-import Popper from '@material-ui/core/Popper'
 import { getCssVariableValue } from '../utils/color'
-import PopperContainerContext from '../PopperContainerContext'
 import Radio from '../Radio'
 import { spacingProp } from '../Stack'
+import { usePopper } from 'react-popper'
+import createDepreciationLogger from '../helpers/createDepreciationLogger'
 
+const logDepecratedProps = createDepreciationLogger()
 const ActionMenuWrapper = ({
   inline,
   onClose,
   anchorElRef,
-  containerElRef,
+  popperOptions,
   placement,
   preventOverflow,
   children
 }) => {
-  const getElementFromRefCallback = ref =>
-    useCallback(() => {
-      return ref ? ref.current : undefined
-    }, [ref])
+  const [popperElement, setPopperElement] = React.useState(null)
+  const referenceElement = anchorElRef ? anchorElRef.current : undefined
 
-  const popperContainerRef = useContext(PopperContainerContext)
-  const getAnchorElement = getElementFromRefCallback(anchorElRef)
-
-  const containerRef = popperContainerRef || containerElRef
-  const getContainerElement = getElementFromRefCallback(containerRef)
-  const normalOverflowModifiers = {
-    preventOverflow: { enabled: false },
-    hide: { enabled: false }
+  const normalOverflowModifiers = [
+    {
+      name: 'preventOverflow',
+      enabled: false
+    },
+    {
+      name: 'hide',
+      enabled: false
+    }
+  ]
+  const options = popperOptions || {
+    placement,
+    modifiers: preventOverflow ? undefined : normalOverflowModifiers
   }
```

Comment:
Why not logging the depreciation notice here instead of the render? (afraid that we flood our dev console )

Reply:
The logger only sends the message once :) https://github.com/cozy/cozy-ui/blob/master/react/helpers/createDepreciationLogger.js#L6 That said I now see I slightly mis-used it.